app-backend/akkautil/src/main/scala/CommonHandlers.scala

```diff
@@ -57,6 +46,29 @@ trait CommonHandlers extends RouteDirectives {
     SecurityDirectives.authorize(user.scope.actions.contains(scopedAction))
   }
 
+  def authorizeScopeLimit(
+      usedLimitFuture: Future[Long],
```

Comment:
I'd 100 percent agree with you on this except in this case we're in `akka-http` land and the signature for `authorizeAsync` requires a future. I think composition of the future for `usedLimitFuture` should be handled outside of this function call -- if it needs to be parallelized or whatever.

Changing it to IO would mean calling `unsafeToFuture` here and requiring a `ContextShift` I think as an implicit argument. Since we haven't done that for other methods here I'm leaning towards not doing that.